### ISSUE TITLE
False positive in lint/assoc-fn when using a macro

### ISSUE THREAD

**Issue by skynet-gh** [NoahTheDuke/splint#15]
Given `splint/bug.clj`:

```clj
(ns splint.bug)

(let [x {}
      y 1]
  (assoc x :a (or (:a x) y)))
```

splint suggests:

```sh
$ clj -M:splint splint/bug.clj
splint/bug.clj:6:3 [lint/assoc-fn] - Use `update` instead of recreating it.
(assoc x :a (or (:a x) y))
Consider using:
(update x :a or y)

Linting took 62ms, checked 1 files, 1 style warnings
```

but this is a syntax error: `Can't take value of a macro: #'clojure.core/or`

**Comment by NoahTheDuke** [comment]
Fixed in release 1.15.1. Thanks for letting me know!
